src/Tablebot/Plugins/Roll/Dice/DiceFunctions.hs

```diff
@@ -91,7 +95,23 @@ funcInfoIndex = FuncInfo "index" [ATInteger, ATIntegerList] ATInteger fiIndex
     fiIndex (LIInteger i : [LIList is])
       | i < 0 || i >= genericLength is = throwBot $ EvaluationException ("index out of range: " ++ show i) []
       | otherwise = return (is !! fromInteger i)
-    fiIndex is = throwBot $ EvaluationException ("incorrect number of arguments. expected 2, got " ++ show (length is)) []
+    fiIndex is = throwBot $ EvaluationException ("incorrect number/type of arguments. expected 2, got " ++ show (length is)) []
+
+funcInfoSet :: FuncInfoBase [Integer]
+funcInfoSet = FuncInfo "set" [ATInteger, ATInteger, ATIntegerList] ATIntegerList fiSet
+  where
+    fiSet (LIInteger i : LIInteger j : [LIList js])
+      | i < 0 || i >= genericLength js = throwBot $ EvaluationException ("index out of range: " ++ show i) []
+      | otherwise = return $ genericTake i js ++ j : genericDrop (i + 1) js
+    fiSet is = throwBot $ EvaluationException ("incorrect number/type of arguments. expected 3, got " ++ show (length is)) []
+
+funcInfoInsert :: FuncInfoBase [Integer]
+funcInfoInsert = FuncInfo "insert" [ATInteger, ATInteger, ATIntegerList] ATIntegerList fiSet
+  where
+    fiSet (LIInteger i : LIInteger j : [LIList js])
+      | i < 0 || i >= genericLength js = throwBot $ EvaluationException ("index out of range: " ++ show i) []
+      | otherwise = return $ genericTake i js ++ j : genericDrop i js
+    fiSet is = throwBot $ EvaluationException ("incorrect number/type of arguments. expected 3, got " ++ show (length is)) []
```

Comment:
When writing helper `funcInfoX` functions, it'd be handy to have a tiny reminder of the function they're providing here (mainly what the inputs are).